src/saml2/config.py

```diff
@@ -569,13 +585,198 @@ def ecp_endpoint(self, ipaddress):
+                self.validate_application_identifier_format(
+                    self.get_application_identifier()),
+            "entityid MUST be an HTTPS URL pointing to the location of its published "
+            "metadata":
+                parse.urlparse(self.entityid).scheme == "https"
+        }
+
+    def validate(self):
+        if not all(self.warning_validators.values()):
+            logger.warning(
+                "Configuration validation warnings occurred: {}".format(
+                    [msg for msg, check in self.warning_validators.items()
+                     if check is not True]
+                )
+            )
+
+        if not all(self.error_validators.values()):
+            error = "Configuration validation errors occurred {}:".format(
+                [msg for msg, check in self.error_validators.items()
+                 if check is not True])
+            logger.error(error)
+            raise ConfigValidationError(error)
+
+
+class eIDASSPConfig(SPConfig, eIDASConfig):
+    def get_endpoint_element(self, element):
+        return getattr(self, "_sp_endpoints", {}).get(element, None)
+
+    def get_application_identifier(self):
+        return getattr(self, "_sp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_sp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_sp_node_country", None)
+
+    @property
+    def warning_validators(self):
+        sp_warning_validators = {
+            "hide_assertion_consumer_service SHOULD be set to True":
+                getattr(self, "_sp_hide_assertion_consumer_service", None) is True
+        }
+        return {**super().warning_validators, **sp_warning_validators}
+
+    @property
+    def error_validators(self):
+        sp_error_validators = {
+            "authn_requests_signed MUST be set to True":
+                getattr(self, "_sp_authn_requests_signed", None) is True,
+            "sp_type MUST be set to 'public' or 'private'":
+                getattr(self, "_sp_sp_type", None) in ("public", "private")
+        }
+        return {**super().error_validators, **sp_error_validators}
+
+
 class IdPConfig(Config):
     def_context = "idp"
 
     def __init__(self):
         Config.__init__(self)
 
 
+class eIDASIdPConfig(IdPConfig, eIDASConfig):
+    def get_endpoint_element(self, element):
+        return getattr(self, "_idp_endpoints", {}).get(element, None)
+
+    def get_application_identifier(self):
+        return getattr(self, "_idp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_idp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_idp_node_country", None)
+
+    def verify_non_notified_loa(self):
+        return not any(
+            [x.startswith("http://eidas.europa.eu/LoA/")
+             for x in getattr(self, "_idp_supported_loa", {}).get("non_notified")]
+        )
```

Comment:
this should return a list if it doesn't find anything:
```python
getattr(self, "_idp_supported_loa", {}).get("non_notified", [])
```

`any` can consume a comprehension directly, without an intermediate list:
```python
any(
    x.startswith("http://eidas.europa.eu/LoA/")
    for x in getattr(self, "_idp_supported_loa", {}).get("non_notified", [])
)
```

Reply:
right :+1:

Reply:
> this should return a list if it doesn't find anything:
> 
> ```python
> getattr(self, "_idp_supported_loa", {}).get("non_notified", [])
> ```
and proper tests would catch this 
fixed now